### PR TITLE
OV-39: integrate Open AI

### PR DESCRIPTION
**Pull request by sergiy4**
This PR adds Open AI integration.
openai1
openai2

### CODE REVIEW

File: backend/src/common/services/open-ai/open-ai.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public async generateText(messages: Message[]): Promise<GeneratedText> {
+        const completion = await this.openAi.chat.completions.create({
+            messages,
+            model: CHAT_MODEL,
+            max_tokens: MAX_TOKEN,
+        });
+
+        const generatedText = completion.choices[0]?.message.content || '';
+        return { generatedText };
```
Comment (Sanchousina) [inline backend/src/common/services/open-ai/open-ai.service.ts:39]: Why not return just a string and not an object with a string, like if even from the name it says it returns text - which is string?

**Review by WilsonBravo — APPROVED** [review]

**Review by o-nedashkivska — APPROVED** [review]

File: backend/src/bundles/open-ai/libs/constants/chat-model.constant.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const CHAT_MODEL = 'gpt-4o-mini' as const;
```
Comment (nikita-remeslov) [inline backend/src/bundles/open-ai/libs/constants/chat-model.constant.ts:1]: as const we use for Enums

File: backend/src/bundles/open-ai/libs/constants/max-token.constant.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const MAX_TOKEN = 2048 as const;
```
Comment (nikita-remeslov) [inline backend/src/bundles/open-ai/libs/constants/max-token.constant.ts:1]: the same

File: backend/src/bundles/open-ai/open-ai.service.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    public async generateText(messages: Message[]): Promise<GeneratedText> {
+        const completion = await this.openAi.chat.completions.create({
+            messages,
+            model: CHAT_MODEL,
+            max_tokens: MAX_TOKEN,
+        });
+
+        const generatedText = completion.choices[0]?.message.content || '';
+        return { generatedText };
```
Comment (nikita-remeslov) [inline backend/src/bundles/open-ai/open-ai.service.ts:39]: why we need return object?

File: backend/src/bundles/open-ai/open-ai.service.ts
Comment (nikita-remeslov) [inline backend/src/bundles/open-ai/open-ai.service.ts:1]: image
looks like it works like this

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by Sanchousina — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**